File: c/hash-table/isomorphic-strings.c
bool isIsomorphic(char* s, char* t) {
    int flag[256] = {0};
    int visited[256] = {0};
    int len = strlen(s);

    for(int i = 0; i < len; ++i){
        if(flag[s[i]] == 0){
            if(visited[t[i]])
                return false;

            flag[s[i]] = t[i];
            visited[t[i]] = 1;
        }else if(flag[s[i]] != t[i]){
            return false;
        }
    }

    return true;
}

File: python/hash-table/isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        slen = len(s)
        flag = {}
        visited = {}

        for i in range(slen):
            if flag.get(s[i],-1) == -1:
                if visited.get(t[i],-1) != -1:
                    return False

                flag[s[i]] = t[i]
                visited[t[i]] = 1
            elif flag.get(s[i]) != t[i]:
                return False

        return True
